Source/Diagnostics/FieldIO.cpp

```diff
@@ -1,9 +1,193 @@
 
 #include <WarpX.H>
 #include <FieldIO.H>
+#ifdef WARPX_USE_OPENPMD
+#include <openPMD/openPMD.hpp>
+#endif
 
 using namespace amrex;
 
+#ifdef WARPX_USE_OPENPMD
+/** \brief For a given field that is to be written to an openPMD file,
+ * set the metadata that indicates the physical unit.
+ */
+void
+setOpenPMDUnit( openPMD::Mesh mesh, const std::string field_name )
+{
+    if (field_name[0] == *"E"){  // Electric field
+        mesh.setUnitDimension({
+            {openPMD::UnitDimension::L,  1},
+            {openPMD::UnitDimension::M,  1},
+            {openPMD::UnitDimension::T, -3},
+            {openPMD::UnitDimension::I, -1},
+        });
+    } else if (field_name[0] == *"B"){ // Magnetic field
+        mesh.setUnitDimension({
+            {openPMD::UnitDimension::M,  1},
+            {openPMD::UnitDimension::I, -1},
+            {openPMD::UnitDimension::T, -2}
+        });
+    } else if (field_name[0] == *"j"){ // current
```

Comment:
shouldn't this be a current density, so `I/L^2`? Or is this a real current `I`? Confused about the dimension below, but maybe its differently normalized.

Reply:
Wow, good catch! Turns out I had it wrong in [warp](https://bitbucket.org/berkeleylab/warp/commits/b6b5aa124f234ee99e022684d62a7ee0b6a178bc) and [fbpic](https://github.com/fbpic/fbpic/commit/51cf7eed08ebabc2638b1b8fd8c1bf1e5d06c5de) as well! Thanks for noticing!